src/state/error.tsx

```diff
@@ -0,0 +1,23 @@
+import { createContext, PropsWithChildren, useState } from 'react'
```

Comment:
I think this logic should be rolled into the `ErrorBoundary`, as they work in tandem. I don't think you need a whole new file to accomplish it.